src/Nethermind/Nethermind.Blockchain/BlockTree.cs

```diff
@@ -555,7 +555,7 @@ private AddBlockResult Suggest(Block block, BlockHeader header, bool shouldProce
                 _headerDb.Set(header.Hash, newRlp.Bytes);
 
                 BlockInfo blockInfo = new BlockInfo(header.Hash, header.TotalDifficulty ?? 0);
-                UpdateOrCreateLevel(header.Number, blockInfo);
+                UpdateOrCreateLevel(header.Number, blockInfo, !shouldProcess);
```

Comment:
interesting - if all tests pass it may be good but would be worth to add a comment why - it is cryptic withou an explanation